src/dev/LCD.cpp

```diff
@@ -255,6 +212,65 @@ void LCD::setTextForSection(uint8_t section, const char* text) {
     sectionColumn += padding;
 
     // Write the text to the screen under the section header.
-    writeText(text, sectionPage, sectionColumn, false);
+    writeText(text, sectionPage, sectionColumn, SMALL, false);
+}
+
+void LCD::writeText(const char* text, uint8_t page, uint8_t column, LCD::FontSize fontSize, bool wrapText) {
+    for (uint8_t x = 0; x < (uint8_t) strlen(text); x++) {
+        if (column >= screenSizeX) {
+            return;
+        }
+
+        // Get the ASCII value of the character.
+        uint8_t characterIndex = text[x];
+        uint8_t characterWidth = fontSize == LARGE ? 8 : 4;
+        uint8_t characterPages = fontSize == LARGE ? 2 : 1;
+
+        switch (fontSize) {
+        case LARGE: {
+            // Create the character that we need to write to the screen.
+            unsigned char characterMap[16] = {
+                BitmapFont::font6x13[characterIndex][0],
+                BitmapFont::font6x13[characterIndex][1],
+                BitmapFont::font6x13[characterIndex][2],
+                BitmapFont::font6x13[characterIndex][3],
+                BitmapFont::font6x13[characterIndex][4],
+                BitmapFont::font6x13[characterIndex][5],
+                BitmapFont::font6x13[characterIndex][6],
+                BitmapFont::font6x13[characterIndex][7],
+                BitmapFont::font6x13[characterIndex][8],
+                BitmapFont::font6x13[characterIndex][9],
+                BitmapFont::font6x13[characterIndex][10],
+                BitmapFont::font6x13[characterIndex][11],
+                BitmapFont::font6x13[characterIndex][12],
+                0b00000000,
+                0b00000000,
+                0b00000000,
+            };
+
+            // Display the character bit map at the calculated page and column.
+            displayBitMapInArea(characterMap, characterWidth, characterPages, page, column);
+        } break;
+        case SMALL: {
+            unsigned char characterMap[4] = {
+                BitmapFont::font4x6[characterIndex][0],
+                BitmapFont::font4x6[characterIndex][1],
+                BitmapFont::font4x6[characterIndex][2],
+                BitmapFont::font4x6[characterIndex][3],
+            };
+
+            // Display the character bit map at the calculated page and column.
+            displayBitMapInArea(characterMap, characterWidth, characterPages, page, column);
```

Comment:
Is the use of characterMap necessary? Could you not just call `displayBitMapInArea(BitmapFont::font4x6[characterIndex], characterWidth, characterPages, page, column);`

Reply:
Thats a good catch, I don't think it is necessary for that part of the function. I think for the `writeLargeText` function it is still going to be necessary because 3 bytes of 0s need to be appended. But if there is some other way to do it I think it would be a good idea